lib/metadataTypes/MetadataType.js

```diff
@@ -566,32 +566,12 @@ class MetadataType {
      * @returns {Promise<Util.MetadataTypeMap>} keyField => metadata map
      */
     static async retrieveSOAPBody(fields, options, type) {
-        let status;
-        let batchCounter = 1;
-        const defaultBatchSize = 2500; // 2500 is the typical batch size
         options = options || {};
-        let metadata = {};
-        do {
-            const resultsBatch = await this.client.soap.retrieve(
-                type || this.definition.type,
-                fields,
-                options
-            );
-            status = resultsBatch.OverallStatus;
-            if (status === 'MoreDataAvailable') {
-                options.continueRequest = resultsBatch.RequestID;
-                Util.logger.info(
-                    `-  more than ${batchCounter * defaultBatchSize} ${
-                        this.definition.typeName
-                    }s found in Business Unit - loading next batch...`
-                );
-                batchCounter++;
-            }
-            const metadataBatch = this.parseResponseBody(resultsBatch);
-
-            metadata = { ...metadata, ...metadataBatch };
-        } while (status === 'MoreDataAvailable');
-
+        const metadata = await this.client.soap.retrieveBulk(
```

Comment:
can you return the promise directly?

Reply:
good point

Reply:
i can do you one better and remove retrieveSOAPBody() all together